File: address-validation-canada-2-nodejs-examples/validate_canadian_municipality_province_rest_sdk_example.js
import { ValidateCanadianMunicipalityProvinceClient } from '../address-validation-canada-2-nodejs/REST/validate_canadian_municipality_province_rest.js';

async function ValidateCanadianMunicipalityProvinceRestGo(licenseKey, isLive) {
    console.log("\n-------------------------------------------------------");
    console.log("AVCA2 - ValidateCanadianMunicipalityProvince - REST SDK");
    console.log("-------------------------------------------------------");

    const municipality = "Bowen Island";
    const province = "BC";
    const postalCode = "V0N 1G2";
    const timeoutSeconds = 15;

    console.log("\n* Input *\n");
    console.log(`Municipality   : ${municipality}`);
    console.log(`Province       : ${province}`);
    console.log(`Postal Code    : ${postalCode}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try {
        const response = await ValidateCanadianMunicipalityProvinceClient.invoke(
            municipality,
            province,
            postalCode,
            licenseKey,
            isLive,
            timeoutSeconds
        );

        if (response.Error) {
            console.log("\n* Error *\n");
            console.log(`Error Type     : ${response.Error.Type}`);
            console.log(`Error Type Code: ${response.Error.TypeCode}`);
            console.log(`Error Desc     : ${response.Error.Desc}`);
            console.log(`Error Desc Code: ${response.Error.DescCode}`);
            return;
        }

        console.log("\n* Address Info *\n");
        if (response) {
            if (response.CanadianAddressInfo) {
                console.log(`Address              : ${response.CanadianAddressInfo.Address}`);
                console.log(`Address2             : ${response.CanadianAddressInfo.Address2}`);
                console.log(`Municipality         : ${response.CanadianAddressInfo.Municipality}`);
                console.log(`Province             : ${response.CanadianAddressInfo.Province}`);
                console.log(`Postal Code          : ${response.CanadianAddressInfo.PostalCode}`);
                console.log(`Time Zone            : ${response.CanadianAddressInfo.TimeZone}`);
                console.log(`Address Number       : ${response.CanadianAddressInfo.AddressNumberFragment}`);
                console.log(`Street Name          : ${response.CanadianAddressInfo.StreetNameFragment}`);
                console.log(`Street Type          : ${response.CanadianAddressInfo.StreetTypeFragment}`);
                console.log(`Directional Code     : ${response.CanadianAddressInfo.DirectionalCodeFragment}`);
                console.log(`Unit Type            : ${response.CanadianAddressInfo.UnitTypeFragment}`);
                console.log(`Unit Number          : ${response.CanadianAddressInfo.UnitNumberFragment}`);
                console.log(`Is PO Box            : ${response.CanadianAddressInfo.IsPOBox}`);
                console.log(`Box Number           : ${response.CanadianAddressInfo.BoxNumberFragment}`);
                console.log(`Station Info         : ${response.CanadianAddressInfo.StationInfo}`);
                console.log(`Delivery Mode        : ${response.CanadianAddressInfo.DeliveryMode}`);
                console.log(`Delivery Installation: ${response.CanadianAddressInfo.DeliveryInstallation}`);
            } else {
                console.log("No CanadianAddressInfo found.");
            }
        } else {
            console.log("No address info found.");
        }
    } catch (e) {
        console.log("\n* Error *\n");
        console.log(`Error Message: ${e.message}`);
    }
}

export { ValidateCanadianMunicipalityProvinceRestGo };

File: address-validation-canada-2-python-examples/validate_canadian_municipality_province_rest_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../address-validation-canada-2-python/REST"))

from validate_canadian_municipality_province_rest import validate_canadian_municipality_province

def validate_canadian_municipality_province_rest_sdk_go(is_live: bool, license_key: str) -> None:
    print("\r\n-------------------------------------------------------")
    print("AVCA2 - ValidateCanadianMunicipalityProvince - REST SDK")
    print("-------------------------------------------------------")

    municipality = "Bowen Island"
    province = "BC"
    postal_code = "V0N 1G2"

    print("\r\n* Input *\r\n")
    print(f"Municipality: {municipality}")
    print(f"Province    : {province}")
    print(f"Postal Code : {postal_code}")
    print(f"License Key : {license_key}")
    print(f"Is Live     : {is_live}")

    try:
        response = validate_canadian_municipality_province(
            municipality, province, postal_code, license_key, is_live
        )

        print("\r\n* Address Info *\r\n")
        if response and not response.Error:
            if response.CanadianAddressInfo:
                address_info = response.CanadianAddressInfo
                print(f"Address              : {address_info.Address }")
                print(f"Address2             : {address_info.Address2 }")
                print(f"Municipality         : {address_info.Municipality}")
                print(f"Province             : {address_info.Province}")
                print(f"Postal Code          : {address_info.PostalCode}")
                print(f"Time Zone            : {address_info.TimeZone}")
                print(f"Address Number       : {address_info.AddressNumberFragment }")
                print(f"Street Name          : {address_info.StreetNameFragment }")
                print(f"Street Type          : {address_info.StreetTypeFragment }")
                print(f"Directional Code     : {address_info.DirectionalCodeFragment }")
                print(f"Unit Type            : {address_info.UnitTypeFragment }")
                print(f"Unit Number          : {address_info.UnitNumberFragment }")
                print(f"Is PO Box            : {address_info.IsPOBox}")
                print(f"Box Number           : {address_info.BoxNumberFragment }")
                print(f"Station Info         : {address_info.StationInfo }")
                print(f"Delivery Mode        : {address_info.DeliveryMode}")
                print(f"Delivery Installation: {address_info.DeliveryInstallation}")
            else:
                print("No CanadianAddressInfo found.")
        else:
            print("No address info found.")

        if response.Error:
            print("\r\n* Error *\r\n")
            print(f"Error Type      : {response.Error.Type}")
            print(f"Error Type Code : {response.Error.TypeCode}")
            print(f"Error Desc      : {response.Error.Desc}")
            print(f"Error Desc Code : {response.Error.DescCode}")

    except Exception as e:
        print("\r\n* Error *\r\n")
        print(f"Error Message: {str(e)}")